source/extensions/filters/http/aws_request_signing/aws_request_signing_filter.cc

```diff
@@ -0,0 +1,71 @@
+#include "extensions/filters/http/aws_request_signing/aws_request_signing_filter.h"
+
+#include "envoy/extensions/filters/http/aws_request_signing/v3/aws_request_signing.pb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace AwsRequestSigningFilter {
+
+FilterConfigImpl::FilterConfigImpl(HttpFilters::Common::Aws::SignerPtr&& signer,
+                                   const std::string& stats_prefix, Stats::Scope& scope)
+    : signer_(std::move(signer)), stats_(Filter::generateStats(stats_prefix, scope)) {}
+
+Filter::Filter(const std::shared_ptr<FilterConfig>& config) : config_(config) {}
+
+HttpFilters::Common::Aws::Signer& FilterConfigImpl::signer() { return *signer_; }
+
+FilterStats& FilterConfigImpl::stats() { return stats_; }
+
+FilterStats Filter::generateStats(const std::string& prefix, Stats::Scope& scope) {
+  const std::string final_prefix = prefix + "aws_request_signing.";
+  return {ALL_AWS_REQUEST_SIGNING_FILTER_STATS(POOL_COUNTER_PREFIX(scope, final_prefix))};
+}
+
+Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool) {
+  try {
+    config_->signer().sign(headers);
+    config_->stats().auth_header_added_.inc();
+  } catch (EnvoyException&) {
+    config_->stats().signing_failed_.inc();
+  }
+
+  return Http::FilterHeadersStatus::Continue;
+}
+
+Http::FilterDataStatus Filter::decodeData(Buffer::Instance&, bool) {
+  return Http::FilterDataStatus::Continue;
+}
```

Comment:
You can derive from PassThroughFilter (IIRC that is the name, check source) to avoid all of this boilerplate.

Reply:
Yep, lets do it.